subprojects/core-api/src/main/java/org/gradle/api/provider/MapProperty.java

```diff
@@ -188,4 +154,40 @@ public interface MapProperty<K, V> extends Provider<Map<K, V>>, HasConfigurableV
      */
     @Override
     void finalizeValue();
+
+    /**
+     * Updates the value of this property in place by executing the provided transformer.
+     * The transformer accepts the frozen value of this property.
+     *
+     * @param transformer the transformer to apply to frozen value of the property
+     * @return this
+     * @since 8.5
+     */
+    @Incubating
+    MapProperty<K, V> update(Transformer<? extends Provider<? extends Map<? extends K, ? extends V>>, ? super Provider<? extends Map<? extends K, ? extends V>>> transformer);
```

Comment:
Does this need to use `Provider`s for the value or just the `Map` itself?  It seems like you can't do anything with a `Provider<Map<...>>` other than getting its value?

Reply:
Sorry for the noise, this change actually was brought in from @mlopatkin's https://github.com/gradle/gradle/pull/26634, please see his RFC on "Handling circular references in lazy configuration API".